### PR TITLE
[t175] do not export generated partition columns by default

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: docs/sql/ddl.txt [hunk 1]
```diff
@@ -193,6 +193,12 @@ compute the value to partition by from existing columns in the table::
     ... ) PARTITIONED BY (partition_value);
     CREATE OK (... sec)
 
+.. note::
+
+   A generated column used as a partition by value, is not exported
```
Comment (dobe) [inline docs/sql/ddl.txt]: if a generated column is a partition column

File: docs/sql/ddl.txt [hunk 2]
```diff
@@ -193,6 +193,12 @@ compute the value to partition by from existing columns in the table::
...
+   using ``copy to`` by default. It must be explicitly added to the
+   list of columns to be included in exported rows.
```
Comment (dobe) [inline docs/sql/ddl.txt]: this changes the output format, so this advice is wrong  see https://crate.io/docs/stable/sql/reference/copy_to.html#parameters

Comment (seut) [inline docs/sql/ddl.txt]: so remove advise or document changed output?

File: sql/src/test/java/io/crate/planner/PlannerTest.java [hunk 3]
```diff
@@ -79,26 +80,38 @@
 
     private Analyzer analyzer;
     private Planner planner;
-    Routing shardRouting = new Routing(TreeMapBuilder.<String, Map<String, List<Integer>>>newMapBuilder()
+    static final Routing SHARD_ROUTING = new Routing(TreeMapBuilder.<String, Map<String, List<Integer>>>newMapBuilder()
```
Comment (dobe) [inline sql/src/test/java/io/crate/planner/PlannerTest.java]: do you know why this is package visible?

Comment (seut) [inline sql/src/test/java/io/crate/planner/PlannerTest.java]: no, think we can change it

File: sql/src/test/java/io/crate/planner/PlannerTest.java [hunk 4]
```diff
@@ -112,6 +125,12 @@ public void prepare() throws Exception {
                 .createInjector();
         analyzer = injector.getInstance(Analyzer.class);
         planner = injector.getInstance(Planner.class);
+
+        if (GENERATED_PARTITIONED_TABLE_INFO == null) {
```
Comment (dobe) [inline sql/src/test/java/io/crate/planner/PlannerTest.java]: this needs to be threadsafe if you do this, just make it non static

**Comment by dobe** [comment]
ack

**Pull request merged**